internal/session/service_worker_status_report.go

```diff
@@ -25,24 +25,34 @@ type StateReport struct {
 func WorkerStatusReport(ctx context.Context, repo *Repository, connRepo *ConnectionRepository, workerId string, report []StateReport) ([]StateReport, error) {
 	const op = "session.WorkerStatusReport"
 
-	reportedConnections := make([]string, 0)
+	reportedConnectionIds := make([]string, 0)
+	reportedConnections := make([]*Connection, 0)
 	reportedSessions := make([]string, 0, len(report))
 	for _, r := range report {
 		reportedSessions = append(reportedSessions, r.SessionId)
-		reportedConnections = append(reportedConnections, r.ConnectionIds...)
+		for _, c := range r.Connections {
+			reportedConnectionIds = append(reportedConnectionIds, c.GetPublicId())
+			reportedConnections = append(reportedConnections, &c)
+		}
 	}
 
 	notActive, err := repo.checkIfNoLongerActive(ctx, reportedSessions)
 	if err != nil {
 		return nil, errors.New(ctx, errors.Internal, op, fmt.Sprintf("Error checking session state for worker %s: %v", workerId, err))
 	}
 
-	closed, err := connRepo.closeOrphanedConnections(ctx, workerId, reportedConnections)
+	closed, err := connRepo.closeOrphanedConnections(ctx, workerId, reportedConnectionIds)
 	if err != nil {
 		return notActive, errors.New(ctx, errors.Internal, op, fmt.Sprintf("Error closing orphaned connections for worker %s: %v", workerId, err))
 	}
 	if len(closed) > 0 {
 		event.WriteSysEvent(ctx, op, "marked unclaimed connections as closed", "controller_id", workerId, "count", len(closed))
 	}
-	return notActive, err
+
+	err = connRepo.updateBytesUpBytesDown(ctx, reportedConnections...)
+	if err != nil {
+		return notActive, errors.New(ctx, errors.Internal, op, fmt.Sprintf("failed to update bytes up and down for worker reported connections: %v", err))
+	}
```

Comment:
I'm concerned that updates to the bytes up and bytes down values might be lost because of a potentially unrelated error being returned from either `checkIfNoLongerActive` or `closeOrphanedConnections`. Can we prevent this from happening?

Reply:
Good catch - I noticed that none of the calls in that function actually depend on each other, so I changed it to use `multierror` and just return error or nil at the end. [c2bf0f3](https://github.com/hashicorp/boundary/pull/2503/commits/c2bf0f309456ba769136c9b27f81c00ecb5f2d89)